From 8432156777da703dc2fea9528da9dade938e6744 Mon Sep 17 00:00:00 2001
From: Paul Gortmaker <paul.gortmaker@windriver.com>
Date: Thu, 11 Aug 2022 14:14:29 -0400
Subject: [PATCH] qemu: vcpu: avoid using ifdef CONFIG_KVM in common code.

CONFIG_KVM should not be used in common code.  If we look at:

commit cbca3722a33c ("include/exec/poison: Mark CONFIG_KVM as poisoned, too")
we see that it states:

    CONFIG_KVM is only defined for target-specific code, so nobody should
    use it by accident in common code. To avoid such subtle bugs,
    CONFIG_KVM is now marked as poisoned in common code.

Since there already is a kvm-stub for when CONFIG_KVM is not enabled,
we can make use of that and make the common code less "ifdeffy" at
the same time.

Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>
---
 accel/stubs/kvm-stub.c | 5 +++++
 softmmu/vl.c           | 2 --
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/accel/stubs/kvm-stub.c b/accel/stubs/kvm-stub.c
index 3345882d85d5..299b8005947a 100644
--- a/accel/stubs/kvm-stub.c
+++ b/accel/stubs/kvm-stub.c
@@ -12,6 +12,7 @@
 
 #include "qemu/osdep.h"
 #include "sysemu/kvm.h"
+#include "sysemu/cpus.h"
 #include "hw/pci/msi.h"
 
 KVMState *kvm_state;
@@ -148,3 +149,7 @@ bool kvm_dirty_ring_enabled(void)
 {
     return false;
 }
+
+void vcpu_parse(const char *optarg)
+{
+}
diff --git a/softmmu/vl.c b/softmmu/vl.c
index 29d178f81245..39c5c4dcc807 100644
--- a/softmmu/vl.c
+++ b/softmmu/vl.c
@@ -3411,9 +3411,7 @@ void qemu_init(int argc, char **argv, char **envp)
                                            "smp", optarg);
                 break;
             case QEMU_OPTION_vcpu:
-#ifdef CONFIG_KVM
                 vcpu_parse(optarg);
-#endif
                 break;
             case QEMU_OPTION_vnc:
                 vnc_parse(optarg);
-- 
2.17.1

